http/analysis-service/server.py

```diff
@@ -846,6 +849,135 @@ def kill(instance_id):
         monitoring_url  = abs_url_for('monitor', instance_id = instance.id)
     )
 
+@app.route("/cluster", methods=["GET"])
+@login_required
+def cluster_get_params(errors=None, values=None):
+    # Check that the user logged in is also authorized to do this
+    if not current_user.is_authorized():
+        return login_manager.unauthorized()
+    return render_template('cluster/cluster.html', errors=errors, values=values,
+        token=str(uuid4()))
+
+@app.route("/cluster/new", methods=["POST"])
+@login_required
+def cluster_spawn():
+    # Check that the user logged in is also authorized to do this
+    if not current_user.is_authorized():
+        return login_manager.unauthorized()
+
+    errors = {}
+
+    # Check required fields
+    for f in ['name', 'token', 'n-workers']:
+        val = request.form[f]
+        if val is None or val.strip() == '':
+            errors[f] = "This field is required"
+
+    if not re.match("\d+", request.form["n-workers"]):
+        errors["n-workers"] = "This field should be a number"
+
+    # Check required file
+    if not request.files['public-ssh-key']:
+        errors['code-tarball'] = "Public key file is required"
+
+    # Bug 961200: Check that a proper OpenSSH public key was uploaded.
+    # It should start with "ssh-rsa AAAAB3"
+    pubkey = request.files['public-ssh-key'].read()
+    if not pubkey.startswith("ssh-rsa AAAAB3") and not pubkey.startswith("ssh-dss AAAAB3"):
+        errors['public-ssh-key'] = "Supplied file does not appear to be a valid OpenSSH public key."
```

Comment:
Please extract a common function for validating the public key (here and in `spawn_worker_instance()`)